### PR TITLE
CHANGELOG: update and include a warning about updating Secondaries.

### PR DESCRIPTION
**Pull request by patrickvacek**
Tagging a few extra people to make sure the text about upgrading is appropriately understandable.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,7 +4,21 @@ This file summarizes notable changes introduced in aktualizr version. It roughly
 
 Our versioning scheme is `YEAR.N` where `N` is incremented whenever a new release is issued. Thus `N` does not necessarily map to months of the year.
 
-## [??? (unreleased)]
+## [upcoming release]
+
+### Special considerations
+
+As a result of changes to the IP/POSIX Secondary protocol (see below), users of these Secondaries will need to take special care when upgrading their devices. The new version of aktualizr (e.g. the Primary) is backwards compatible and will work with both old and new versions of the protocol. However, the aktualizr-secondary is *not*. This means that if you are upgrading a device with IP/POSIX Secondaries, you should update the Primary ECU first, and if that is successful, then update your Secondaries.
```
Comment (HalynaDumych) [inline CHANGELOG.md:11]:
```suggestion
As a result of changes to the IP/POSIX Secondary protocol (see below), users of these Secondaries will need to take special care when upgrading their devices. The new version of aktualizr (for example, the Primary) is backwards compatible and will work with both old and new versions of the protocol. However, the aktualizr-secondary is *not*. This means that if you are upgrading a device with IP/POSIX Secondaries, you should update the Primary ECU first, and if that is successful, update your Secondaries.
```

Comment (HalynaDumych) [reply CHANGELOG.md:11]: `e.g.` is correct, but as a general consideration, we should omit Latin words in the documentation.

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,7 +4,21 @@ This file summarizes notable changes introduced in aktualizr version. It roughly
...
+
+### Added
+
+- SetInstallationRawReport API call to enable setting a custom raw report field in the device installation result: [PR](https://github.com/advancedtelematic/aktualizr/pull/1628)
+- Allow ECU re-registration to support adding, removing, and replacing Secondaries: [PR](https://github.com/advancedtelematic/aktualizr/pull/1686)
+
+### Changed
+
+- Improved Secondary interface and error reporting, and improved IP/POSIX Secondary protocol: [PR](https://github.com/advancedtelematic/aktualizr/pull/1642)
```
[the remark this answers is not in the document]
Comment (patrickvacek) [reply CHANGELOG.md:20]: This is complicated, because there are three items that I'm trying to be clear about:
* the Secondary interface
* Secondary error reporting
* the IP/POSIX Secondary protocol

The problem is that the first two are general and affect all Secondaries, while the third does not. Maybe I'll just list them as two separate entries to make the distinction clearer.

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,7 +4,21 @@ This file summarizes notable changes introduced in aktualizr version. It roughly
...
+## [upcoming release]
+
+### Special considerations
+
+As a result of changes to the IP/POSIX Secondary protocol (see below), users of these Secondaries will need to take special care when upgrading their devices. The new version of aktualizr (e.g. the Primary) is backwards compatible and will work with both old and new versions of the protocol. However, the aktualizr-secondary is *not*. This means that if you are upgrading a device with IP/POSIX Secondaries, you should update the Primary ECU first, and if that is successful, then update your Secondaries.
+
+### Added
+
+- SetInstallationRawReport API call to enable setting a custom raw report field in the device installation result: [PR](https://github.com/advancedtelematic/aktualizr/pull/1628)
+- Allow ECU re-registration to support adding, removing, and replacing Secondaries: [PR](https://github.com/advancedtelematic/aktualizr/pull/1686)
```
[the remark this answers is not in the document]
Comment (patrickvacek) [reply CHANGELOG.md:16]: I don't think I agree with this one. I want to stress that this feature is now allowed going forward. We generally use the present tense for anything that is added like this. The past tense is only used for actions that were taken once, usually referring specifically to the code itself. I see that we aren't consistent on that, but present tense is much clearer here IMO.

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,7 +4,21 @@ This file summarizes notable changes introduced in aktualizr version. It roughly
...
-## [??? (unreleased)]
+## [upcoming release]
+
+### Special considerations
+
+As a result of changes to the IP/POSIX Secondary protocol (see below), users of these Secondaries will need to take special care when upgrading their devices. The new version of aktualizr (e.g. the Primary) is backwards compatible and will work with both old and new versions of the protocol. However, the aktualizr-secondary is *not*. This means that if you are upgrading a device with IP/POSIX Secondaries, you should update the Primary ECU first, and if that is successful, then update your Secondaries.
+
+### Added
+
+- SetInstallationRawReport API call to enable setting a custom raw report field in the device installation result: [PR](https://github.com/advancedtelematic/aktualizr/pull/1628)
```
[the remark this answers is not in the document]
Comment (patrickvacek) [reply CHANGELOG.md:15]: I'm not sure about either of these. We added a new API function, and since this is the "Added" section, we aren't strict about complete sentences when it is clear the described feature was added. Neither of the suggestions are accurate, unfortunately.

File: CHANGELOG.md (hunk 1 above)
Comment (patrickvacek) [reply CHANGELOG.md:11]: Actually it turns out I meant "i.e." I often mix those up. In this case, I meant that aktualizr = the Primary. I think I see how I can rephrase to make it clearer.

Comment (patrickvacek) [reply CHANGELOG.md:11]: Also: why remove the "then"? I don't think it is strictly necessary, but I think the if-then construction is clearer than just "if" alone.

Comment (HalynaDumych) [reply CHANGELOG.md]: i.e. => that is

As to "then", yep, it's not strictly necessary. It's also clear without "then". It was just a small suggestion to make the sentence shorter. But if you decide to go for "if-then", I'll have nothing against that :)

File: CHANGELOG.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply CHANGELOG.md:15]: Looks like we can read only this sentence as a continuation to the section heading. It may be a bit misleading to users.

File: CHANGELOG.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply CHANGELOG.md:16]: I see your point. Can we maybe rephrase it?
```
- ECU re-registration now supports adding, removing, and replacing Secondaries: [PR](https://github.com/advancedtelematic/aktualizr/pull/1686)
```

[the remark this answers is not in the document]
Comment (patrickvacek) [reply CHANGELOG.md:16]: It's tricky. Your suggestion isn't completely accurate, because ECU re-registration wasn't possible before. I'm a bit stuck on how to improve this while remaining technically precise.

File: CHANGELOG.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (patrickvacek) [reply CHANGELOG.md:15]: Yes, I see your point. But we've been doing that since the beginning and I'm not sure how to make it flow nicely without repeating "Added" in most of the entries.

File: CHANGELOG.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply CHANGELOG.md:16]: Aha, I misunderstood it, sorry.
What about this one?
```
Now, you can re-register ECUs to support adding, removing, and replacing Secondaries: [PR](https://github.com/advancedtelematic/aktualizr/pull/1686)
```

File: CHANGELOG.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply CHANGELOG.md:15]: What about this one?
```
- Now, you can use the `SetInstallationRawReport` API call to enable setting a custom raw report field in the device installation result: [PR](https://github.com/advancedtelematic/aktualizr/pull/1628)
```

File: CHANGELOG.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (patrickvacek) [reply CHANGELOG.md]: Yep, that works. I took the idea and expanded on it just a bit.

File: CHANGELOG.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (patrickvacek) [reply CHANGELOG.md]: Sure. I took this and simplified it just a bit.

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,7 +4,22 @@ This file summarizes notable changes introduced in aktualizr version. It roughly
 
 Our versioning scheme is `YEAR.N` where `N` is incremented whenever a new release is issued. Thus `N` does not necessarily map to months of the year.
 
-## [??? (unreleased)]
+## [upcoming release]
+
+### Special considerations
+
+As a result of changes to the IP/POSIX Secondary protocol (see below), users of these Secondaries will need to take special care when upgrading their devices. The new version of aktualizr is backwards compatible and will work with both old and new versions of the protocol. However, aktualizr-secondary is *not*. This means that if you are upgrading a device with IP/POSIX Secondaries, you should update the Primary ECU running aktualizr first, and if that is successful, then update your Secondaries.
```
Comment (xaviergully) [inline CHANGELOG.md:11]:
```suggestion
As a result of changes to the IP/POSIX Secondary protocol (see below), users of these Secondaries will need to take special care when upgrading their devices. The new version of aktualizr is backwards compatible and will work with both old and new versions of the protocol. However, aktualizr-secondary is *not*. This means that if you are upgrading a device with IP/POSIX Secondaries, you should update the Primary ECU running aktualizr **first**, and if that is successful, then update your Secondaries.
```
